### ISSUE TITLE
AttributeError: module 'cryptography.hazmat.bindings._rust.openssl' has no attribute 'hashes'

### ISSUE THREAD

**Issue by HanumanthMaldakantiTR** [pyca/cryptography#8995]
We are facing issue while importing jwt since this morning (May 31st 2023), 

can you please help with the issue.
```
    import jwt
 File "/opt/python/lib/python3.8/site-packages/jwt/__init__.py", line 19, in <module>
    from .api_jwt import (
  File "/opt/python/lib/python3.8/site-packages/jwt/api_jwt.py", line 11, in <module>
    from .api_jws import PyJWS
  File "/opt/python/lib/python3.8/site-packages/jwt/api_jws.py", line 10, in <module>
    from .algorithms import (
  File "/opt/python/lib/python3.8/site-packages/jwt/algorithms.py", line 15, in <module>
    from cryptography.hazmat.primitives import hashes
  File "/opt/python/lib/python3.8/site-packages/cryptography/hazmat/primitives/hashes.py", line 88, in <module>
    Hash = rust_openssl.hashes.Hash
```

**Comment by reaperhulk** [comment]
Please provide a way to reproduce this in a fresh virtual environment so we can help debug. In the past similar issues have occurred when users have multiple versions of cryptography installed in different paths.

**Comment by Kritish100** [comment]
Hi @HanumanthMaldakantiTR , have you resolved this issue ? can you share the solution please if resolved

**Comment by pattty847** [comment]
Hi, I've noticed this issue arises when I use the library CCXT. I'm sorry if this is not the library OP was talking about, but I found this thread while searching for fixes.

I will install a fresh virtual environment, install the ccxt library, and everything works fine. When I shut my laptop and left the place I was at (left Starbucks came home), and tried running the script again I was confronted with the error you all are talking about. 

Installing a fresh virtual environment does fix the bug, but I can't do it every time. 

Let me know if I can provide any further information. Thank you all a lot.

`Traceback (most recent call last):
  File "/Users/Pat/Desktop/Programming/ccxt testing/main.py", line 1, in <module>
    import ccxt
  File "/Users/Pat/Desktop/Programming/ccxt testing/venv/lib/python3.11/site-packages/ccxt/__init__.py", line 29, in <module>
    from ccxt.base.exchange import Exchange                     # noqa: F401
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Pat/Desktop/Programming/ccxt testing/venv/lib/python3.11/site-packages/ccxt/base/__init__.py", line 24, in <module>
    from ccxt.base import exchange
  File "/Users/Pat/Desktop/Programming/ccxt testing/venv/lib/python3.11/site-packages/ccxt/base/exchange.py", line 38, in <module>
    from cryptography.hazmat.primitives import hashes
  File "/Users/Pat/Desktop/Programming/ccxt testing/venv/lib/python3.11/site-packages/cryptography/hazmat/primitives/hashes.py", line 88, in <module>
    Hash = rust_openssl.hashes.Hash
           ^^^^^^^^^^^^^^^^^^^
AttributeError: module 'cryptography.hazmat.bindings._rust.openssl' has no attribute 'hashes'`

**Comment by reaperhulk** [comment]
This issue can effectively only happen when Python is loading two different copies of cryptography (one of which is older and does not have that import), so there’s likely something else happening in your environment.

**Comment by pattty847** [comment]
> 

Okay, it also only happens on my Mac. I will look into that, thank a lot for your response.

**Comment by gyang909** [comment]
I have the exact same error message as the OP when I test my Lambda function at AWS Lambda console. 

I have Python 3.10 in my local, same as AWS Lambda runtime. I created a local venv. When I run my Python code from the venv, I have no problems. The same venv is packed as a .zip file and deployed to AWS Lambda console, but test from AWS Lambda gives me this error. Doesn't AWS Lambda console use the same venv that I provided to run Lambda? Hope someone can advise.
